src/color.rs

```diff
@@ -385,17 +385,29 @@ impl<T: Primitive> FromPrimitive<T> for T {
 // Note that in to-integer-conversion we are performing rounding but NumCast::from is implemented
 // as truncate towards zero. We emulate rounding by adding a bias.
 
+// All other special values are clamped inbetween 0.0 and 1.0 (infinities and subnormals)
+// NaN however always maps to NaN therefore we have to force it towards some value.
+// 1.0 (white) was picked as firefox and chrome choose to map NaN to that.
+#[inline]
+fn sanitize_nan(i: f32) -> f32 {
+    if i.is_nan() {
+        1.0
+    } else {
+        i
+    }
+}
+
 impl FromPrimitive<f32> for u8 {
     fn from_primitive(float: f32) -> Self {
         let inner = (float.clamp(0.0, 1.0) * u8::MAX as f32).round();
-        NumCast::from(inner).unwrap()
+        NumCast::from(sanitize_nan(inner)).unwrap()
```

Comment:
Wait, shouldn't the `sanitize_nan` call happen before scaling by `u8::MAX`?

Reply:
Good point, ill fix that in a sec